sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionPump.cs

```diff
@@ -136,7 +136,10 @@ public async Task StartAsync()
                         // and let the event processor handle it.  The inner consumer hasn't connected to the service yet,
                         // so there's no need to close it.
 
-                        await PartitionProcessor.InitializeAsync(Context).ConfigureAwait(false);
+                        if (OwnerEventProcessor.InitializeProcessingForPartitionAsync != null)
```

Comment:
We're guaranteed that nobody can change that handler in this section because the processor is active, right?

Reply:
Exactly. With the latest change where I moved the `RunningTask = null` to the `finally` block, we might encounter an issue if we fail in closing a pump. I'll revisit that before merging this PR.

https://github.com/Azure/azure-sdk-for-net/blob/3a7af6961f39e0422e50abc9e92fafb02d7a7860/sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs#L345-L349

Reply:
Just double checked and I'm just worrying too much. It's good the way it is.